data/src/main/java/org/apache/iceberg/data/GenericAppenderFactory.java

```diff
@@ -95,4 +115,88 @@ public GenericAppenderFactory setAll(Map<String, String> properties) {
       throw new UncheckedIOException(e);
     }
   }
+
+  @Override
+  public org.apache.iceberg.io.DataWriter<Record> newDataWriter(EncryptedOutputFile file, FileFormat format,
+                                                                StructLike partition) {
+    return new org.apache.iceberg.io.DataWriter<>(
+        newAppender(file.encryptingOutputFile(), format), format,
+        file.encryptingOutputFile().location(), spec, partition, file.keyMetadata());
+  }
```

Comment:
Could this be the default implementation of `newDataWriter`? It looks the same across all of the implementations.

Reply:
I did not move it to be an default implementation for `newDataWriter`  because the `DataWriter` construction will need the the partition `spec`.  It's not friendly to expose an extra method `spec()` in `FileAppenderFactory` ( so that those subclass could pass the correct `spec` object to `DataWriter`).

Reply:
Sounds good, thanks for explaining it to me.